src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FailOverSupportedConfigurationClient.cs

```diff
@@ -0,0 +1,163 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+
+using Azure;
+using Azure.Data.AppConfiguration;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration.Extensions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
+{
+    internal class FailOverSupportedConfigurationClient : IConfigurationClient
+    {
+        private const int HttpStatusRequestThrottled = 429;
+
+        private readonly IEnumerable<LocalConfigurationClient> clients;
+
+        private DateTimeOffset retryPrimaryConfigClientAfter;
+
+        private int failedRequestsToPrimaryConfigClient = 0;
+
+        public FailOverSupportedConfigurationClient(IEnumerable<LocalConfigurationClient> clients)
+        {
+            if (clients == null)
+            {
+                throw new ArgumentNullException(nameof(clients));
+            }
+
+            if (clients.Count() < 1)
+            {
+                throw new ArgumentException($"{nameof(clients)} needs at least one ClientConfiguration.");
+            }
```

Comment:
These can be combined into one `if` block for simplicity:
`if (clients == null || clients.Count() == 0)`

Reply:
I was just trying to be more explicit with the error message, but it is fine to remove it.